Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -325,11 +325,34 @@ def check_binary_arches(formula)
     end
     return if mismatches.empty?
 
-    <<~EOS
-      Binaries built for a non-native architecture were installed into #{formula}'s prefix.
-      The offending files are:
-        #{mismatches * "\n        "}
-    EOS
+    compatible_universal_binaries = mismatches.select do |file|
+      file.arch == :universal && file.archs.include?(Hardware::CPU.arch)
+    end
+    mismatches -= compatible_universal_binaries
+
+    universal_binaries_expected =
+      formula.tap.present? && tap_audit_exception(:universal_binary_allowlist, formula.name)
+    return if mismatches.empty? && universal_binaries_expected
+
+    s = ""
+
+    unless mismatches.empty?
```

Comment:
```suggestion
    if mismatches.blank?
```

and could use this a few places elsewhere

Reply:
Don't quite follow here, as `mismatches.empty? == mismatches.blank?`. I suppose you're looking for
```ruby
if mismatches.any?
```
?

Reply:
Sorry, I meant `.present?` instead of `unless ... empty?`

Reply:
Gotcha, done.